presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastoreConfig.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.metastore.thrift;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.Duration;
+
+import javax.validation.constraints.NotNull;
+
+import java.util.concurrent.TimeUnit;
+
+public class ThriftHiveMetastoreConfig
+{
+    private int maxRetries = 10;
+    private double backoffScaleFactor = 2.0d;
+    private Duration minBackoffDelay = new Duration(1, TimeUnit.SECONDS);
+    private Duration maxBackoffDelay = new Duration(1, TimeUnit.SECONDS);
+    private Duration maxRetryTime = new Duration(30, TimeUnit.SECONDS);
```

Comment:
Now we have 2 sources of defaults. -- `RetryDriver` and this config class -- whereas we have no intention to do so.

Please

1. Make `RetryDriver` `DEFAULT_*` fields `public`
2. use them here as initial value

Reply:
@findepi  Thanks for your comment;
I have a concerns about the default value of 

`private int maxRetries = RetryDriver.DEFAULT_RETRY_ATTEMPTS - 1;`

Because the maxRetries should just retry, not include the first attempt; From the code of RetryDriver.run:

```

    public <V> V run(String callableName, Callable<V> callable)
            throws Exception
    {
        requireNonNull(callableName, "callableName is null");
        requireNonNull(callable, "callable is null");

        List<Throwable> suppressedExceptions = new ArrayList<>();
        long startTime = System.nanoTime();
        int attempt = 0;
        while (true) {
            attempt++;

            try {
                return callable.call();
            }
            catch (Exception e) {
                if (attempt >= maxAttempts || Duration.nanosSince(startTime).compareTo(maxRetryTime) >= 0) {
                    addSuppressed(e, suppressedExceptions);
                    throw e;
                }
```
we could see that the `RetryDriver.DEFAULT_RETRY_ATTEMPTS` actually at as the **max attempts**(include the first attempt) instead of **max retry attempts**;

Reply:
> `private int maxRetries = RetryDriver.DEFAULT_RETRY_ATTEMPTS - 1;`

that's OK.

> we could clearly see that the `RetryDriver.DEFAULT_RETRY_ATTEMPTS` actually at as the **max attempts**(include the first attempt) instead of **max retry attempts**; So I think naming of RetryDriver.DEFAULT_RETRY_ATTEMPTS is a little irregular;

please rename `DEFAULT_RETRY_ATTEMPTS` to `DEFAULT_MAX_ATTEMPTS`

Reply:
OK